src/main/java/com/mastery/java/task/dao/EmployeeDaoImpl.java

```diff
@@ -0,0 +1,190 @@
+            }
+
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                resultSet.close();
+                preparedStatement.close();
+                con.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+        return employee;
+    }
+
+
+    public void update(Employee employee) {
+        String query = "UPDATE employee set first_name=?, last_name=?, departament_id=?, job_title=?, gender=?, date_of_birth=? where employee_id=?";
+        Connection con = null;
+        PreparedStatement preparedStatement = null;
+        try {
+            con = dataSource.getConnection();
+            preparedStatement = con.prepareStatement(query);
+            preparedStatement.setString(1, employee.getFirstName());
+            preparedStatement.setString(2, employee.getLastName());
+            preparedStatement.setInt(3, employee.getDepartamentId());
+            preparedStatement.setString(4, employee.getJobTitle());
+            preparedStatement.setString(5, String.valueOf(employee.getGender()));
+            preparedStatement.setDate(6, (Date) employee.getDateOfBirth());
+            preparedStatement.setInt(7, employee.getEmployeeId());
+            int out = preparedStatement.executeUpdate();
+            if (out != 0) {
+                System.out.println("Update Employee with id=" + employee.getEmployeeId());
+            } else System.out.println("Employee update failed with id=" + employee.getEmployeeId());
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                preparedStatement.close();
+                con.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+
+    public void deleteById(int id) {
+        String query = "DELETE from employee where employee_id=?";
+        Connection con = null;
+        PreparedStatement preparedStatement = null;
+        try {
+            con = dataSource.getConnection();
+            preparedStatement = con.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            int out = preparedStatement.executeUpdate();
+            if (out != 0) {
+                System.out.println("Employee deleted with id=" + id);
+            } else System.out.println("No Employee found with id=" + id);
+        } catch (SQLException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                preparedStatement.close();
+                con.close();
+            } catch (SQLException e) {
+                e.printStackTrace();
+            }
+        }
+
+    }
+
+
+    public List<Employee> getAll() {
+        String query = "SELECT employee_id, first_name, last_name, departament_id, job_title, gender, date_of_birth from employee";
```

Comment:
Please use named constant String in the top of the class to store SQL query.
Could we simplify it, if it will look like `SELECT * from employee;` ?

Reply:
Done, done, also done * for getById